expenses-manager-backend/src/main/java/com/endava/expensesmanager/service/impl/ExpenseServiceImpl.java

```diff
@@ -0,0 +1,54 @@
+package com.endava.expensesmanager.service.impl;
+
+import com.endava.expensesmanager.dto.ExpenseDto;
+import com.endava.expensesmanager.entity.Expense;
+import com.endava.expensesmanager.mapper.ExpenseMapper;
+import com.endava.expensesmanager.repository.ExpenseRepository;
+import com.endava.expensesmanager.service.ExpenseService;
+import org.springframework.stereotype.Service;
+
+import java.util.List;
+import java.util.Optional;
+
+@Service
+public class ExpenseServiceImpl implements ExpenseService {
+
+    private final ExpenseRepository expenseRepository;
+    private static final ExpenseMapper expenseMapper = ExpenseMapper.INSTANCE;
+
+    public ExpenseServiceImpl(ExpenseRepository expenseRepository) {
+        this.expenseRepository = expenseRepository;
+    }
+
+    @Override
+    public ExpenseDto createExpense(ExpenseDto expenseDto) {
+        Expense createdExpense = expenseMapper.expenseDtoToExpense(expenseDto);
+        createdExpense = expenseRepository.save(createdExpense);
+        return expenseMapper.expenseToExpenseDto(createdExpense);
+    }
+
+    @Override
+    public List<ExpenseDto> getAllExpenses() {
+        List<Expense> expenses = expenseRepository.findAll();
+        return expenses.stream().map(expenseMapper::expenseToExpenseDto).toList();
```

Comment:
I suggest formatting the code to make it easier to read it. Here are some examples https://medium.com/javarevisited/10-java-stream-tips-must-read-2063a84af3be